source/pinia/statistics-store.ts

```diff
@@ -14,17 +14,80 @@
 
 import { type Stats } from '@providers/stats'
 import { defineStore } from 'pinia'
-import { ref } from 'vue'
+import { anyLast30Days, avgAny30Days, sumAny30Days } from 'source/common/util/stats'
+import { DateTime } from 'luxon'
+import { ref, computed } from 'vue'
 
 const ipcRenderer = window.ipc
 
+/**
+ * Generates the past 30 days as ISO dates in the form YYYY-MM-DD
+ *
+ * @return  {string[]}  The date strings
+ */
+function generateLast30ISODates (): string[] {
```

Comment:
I wouldn't risk it, since it relies on Luxon, which in turn relies on a ton of web APIs that aren't necessarily guaranteed to be compiled into Node. We've had this issue already. This is why main will only modify the stats as soon as new words/characters/pomodoros are coming in, since the basic Date functionality used to generate today's `yyyy-mm-dd` will definitely work. This whole generation stuff is only done on the renderer side. The statistics utilities, however, have zero dependencies and will work both in a browser and in a Node process.

This may change in the future, but I've burned my fingers with this once…